spyder/api/plugins/new_api.py

```diff
@@ -236,119 +406,153 @@ class SpyderPluginV2(QObject, SpyderActionMixin, SpyderConfigurationObserver,
-
-    Notes
-    -----
-    To be used by plugins tracking main window size changes.
     """
 
     sig_mainwindow_moved = Signal("QMoveEvent")
     """
-    This signal is emitted when the main window is moved.
+    Emitted when the main window is moved.
+
+    Used by plugins to track main window position changes.
 
     Parameters
     ----------
     move_event: QMoveEvent
         The event triggered on main window move.
-
-    Notes
-    -----
-    To be used by plugins tracking main window position changes.
     """
 
     sig_unmaximize_plugin_requested = Signal((), (object,))
     """
-    This signal is emitted to inform the main window that it needs to
-    unmaximize the currently maximized plugin, if any.
+    Request the main window unmaximize the currently maximized plugin, if any.
 
     Parameters
     ----------
     plugin_instance: SpyderDockablePlugin
-        Unmaximize plugin only if it is not `plugin_instance`.
+        Unmaximize current plugin only if it is not ``plugin_instance``.
     """
 
     sig_mainwindow_state_changed = Signal(object)
     """
-    This signal is emitted when the main window state has changed (for
-    instance, between maximized and minimized states).
+    Emitted when the main window state has changed (e.g. maximized/minimized).
 
     Parameters
     ----------
     window_state: Qt.WindowStates
-        The window state.
+        The new main window state.
     """
 
     sig_focused_plugin_changed = Signal(object)
     """
-    This signal is emitted when the plugin with keyboard focus changes.
+    Emitted when the plugin with keyboard focus changes.
 
     Parameters
     ----------
-    plugin: Optional[SpyderDockablePlugin]
-        The plugin that currently has keyboard focus, or None if no dockable
-        plugin has focus.
+    plugin: SpyderDockablePlugin | None
+        The plugin that currently has keyboard focus, or ``None`` if no
+        dockable plugin has focus.
     """
 
     # ---- Private attributes
     # -------------------------------------------------------------------------
-    # Define configuration name map for plugin to split configuration
-    # among several files. See spyder/config/main.py
+
     _CONF_NAME_MAP = None
+    """
+    Define config name map for plugin to split config among several files.
 
-    def __init__(self, parent, configuration=None):
+    See :mod:`spyder.config.main`.
+    """
+
+    # ---- Private methods
+    # -------------------------------------------------------------------------
+
+    def __init__(
+        self,
+        parent: spyder.app.mainwindow.MainWindow,
+        configuration: (
+            spyder.config.manager.ConfigurationManager | None
+        ) = None,
+    ):
+        """
+        Initialize the plugin object (called automatically by Spyder).
+
+        .. important::
+
+            Plugins are initialized automatically by Spyder; plugins shouldn't
+            call or override this method directly.
```

Comment:
```suggestion
            Plugins are initialized automatically by Spyder, so they shouldn't
            call or override this method directly.
```

Make this sentence flow better.